### ISSUE TITLE
Is page template file applied FileName rules?

### ISSUE THREAD

**Issue by yousan** [WordPress-Coding-Standards/WordPress-Coding-Standards#959]
I got WordPress.Files.FileName.NotHyphenatedLowercase error at page-slug_for_file.php.

```
 1 | ERROR | Filenames should be all lowercase with hyphens as word
   |       | separators. Expected page-slug-for-file.php., but found
   |       | page-slug_for_file.php..php.
   |       | (WordPress.Files.FileName.NotHyphenatedLowercase)
```

`page-$slug.php` is a member of template hierarchy, is it allowed with underscores?

https://developer.wordpress.org/files/2014/10/template-hierarchy.png

is_theme property is not worked for this case. 

https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Customizable-sniff-properties#themes-allow-filename-exceptions

```php
	const THEME_EXCEPTIONS_REGEX = '`
		^                    # Anchor to the beginning of the string.
		(?:
			(?:archive|content|embed|single|taxonomy) # Template prefixes which can have exceptions
			-[^\.]+          # These need to be followed by a dash and some chars.
		|
			(?:application|audio|example|image|message|model|multipart|text|video) #Top-level mime-types
			(?:_[^\.]+)?     # Optionally followed by an underscore and a sub-type.
		)\.(?:php|inc)$      # End in .php (or .inc for the test files) and anchor to the end of the string.
	`Dx';
```

at CodeSniffer/Standards/WordPress/WordPress/Sniffs/Files/FileNameSniff.php

**Comment by jrfnl** [comment]
Thanks for reporting this. I've updated the list of exclusions based on the hierarchy chart. See #960
